From 6b809631d49eddbe2a6efe3b15559e81d6a0bbab Mon Sep 17 00:00:00 2001
From: Phillip Lougher <phillip@squashfs.org.uk>
Date: Sun, 31 Aug 2014 03:47:50 +0100
Subject: [PATCH 085/120] action: expression logging should be using
 action->args

Fix display of the arguments in the action() statement.
Previously it used action->action->args which is the arg
count in the "generic" action statement, not the number of
actual arguments in the particular parsed action - i.e. the instance
rather than the definition.

For actions which take a fixed number of arguments there, obviously,
is no difference.  But for actions which take a variable number of
arguments the generic action count is "-2", and it is the
responsibility of an specific action parser helper to parse
the arguments.  In this case the discovered number of arguments
in the instance will be different to the -2 argument count in the
generic statement.

Signed-off-by: Phillip Lougher <phillip@squashfs.org.uk>
---
 squashfs-tools/action.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/squashfs-tools/action.c b/squashfs-tools/action.c
index c442231..8e615c7 100644
--- a/squashfs-tools/action.c
+++ b/squashfs-tools/action.c
@@ -754,11 +754,11 @@ static int eval_expr_top(struct action *action, struct action_data *action_data)
 		expr_log("=");
 		expr_log(action->action->name);
 
-		if(action->action->args) {
+		if(action->args) {
 			expr_log("(");
-			for (n = 0; n < action->action->args; n++) {
+			for (n = 0; n < action->args; n++) {
 				expr_log(action->argv[n]);
-				if(n + 1 < action->action->args)
+				if(n + 1 < action->args)
 					expr_log(",");
 			}
 			expr_log(")");
@@ -3046,13 +3046,13 @@ void dump_action_list(struct action *spec_list, int spec_count)
 
 	for (i = 0; i < spec_count; i++) {
 		printf("%s", spec_list[i].action->name);
-		if (spec_list[i].action->args) {
+		if (spec_list[i].args) {
 			int n;
 
 			printf("(");
-			for (n = 0; n < spec_list[i].action->args; n++) {
+			for (n = 0; n < spec_list[i].args; n++) {
 				printf("%s", spec_list[i].argv[n]);
-				if (n + 1 < spec_list[i].action->args)
+				if (n + 1 < spec_list[i].args)
 					printf(",");
 			}
 			printf(")");
-- 
2.8.3

